app/models/actor.rb

```diff
@@ -0,0 +1,8 @@
+class Actor < ApplicationRecord
+  has_many :movie_actors
+  has_many :movies, through: :movie_actors
+
+  def self.average_age
+    self.average(:age).round
```

Comment:
well done encapsulating this logic in a model method! However, this is going to get the average age of *all* actors rather than just the actors in a given movie. We were looking for this to be an instance method on the movie model to average just the actors associated with that movie (or you could pass in a movie_id here)

Reply:
When I tested this it was for just the shrek movie, and the average age of the actors for that is 30, but the average age for all of the actors was 40...am I missing something? I added another test with a different movie and both tests passed.

Reply:
Ahh that's because you're calling the method on the collection of actors in the view: `@movie.actors.average_age` so this is actually fine if you call it on the movie instance because it will be bound to just actors in that movie. Either way works. I missed that this was how you were using the method in the view - my bad!

Reply:
If you had done `Actor.average_age` in the view, you would have had an incorrect answer. Still be sure to model test this method though!